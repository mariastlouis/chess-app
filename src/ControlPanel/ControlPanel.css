@import url("https://fonts.googleapis.com/css?family=Pathway+Gothic+One");
@import url("https://fonts.googleapis.com/css?family=Cabin");
body {
  font-family: "Cabin", sans-serif; }

h1, h2, p, label {
  color: #575b57; }

label {
  font-size: 20px; }

.logo {
  width: 100%; }

.radio {
  padding: 3px; }

.controlPanel {
  display: flex;
  flex-direction: column; }

@media screen and (max-width: 1000px) {
  .controlPanel {
    flex-direction: row; } }

@media screen and (max-width: 700px) {
  .controlPanel {
    flex-direction: column; } }
