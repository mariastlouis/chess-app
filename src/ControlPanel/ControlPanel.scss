@import 'variable';

.logo {
  width: 100%
}

.radio {
  padding: 3px;
}

.controlPanel {
  display:flex;
  flex-direction: column;
}

@media screen and (max-width: 1000px) {
 .controlPanel {
  flex-direction: row;
 }
}

@media screen and (max-width: 700px) {
 .controlPanel {
  flex-direction: column;
 }
}
