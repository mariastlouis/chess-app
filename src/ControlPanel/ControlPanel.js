import React, { Component } from 'react';
import PropTypes from 'prop-types';
import logo from '../resources/logo.png'
import './ControlPanel.css'



class ControlPanel extends Component {
  constructor() {
    super()
    this.state = {
      selectedOption: 'standard'
    }
  }



  selectedPieceMessage() {
    if (this.props.chessPiece === 'knight') {
      return 'Knight selected'
    } else if (this.props.chessPiece === 'bishop') {
      return 'Bishop selected'
    } else {
      return ''
    }
  }

  handleOptionChange = event => {
    let mode = event.target.value

    this.setState({selectedOption: mode})
    this.props.mode(mode)

  }

  render () {
  return (
    <div className = "controlPanel">
      <div className = "logo-div sider-div">
        <img src = {logo} alt = "logo" className = "logo"/>
      </div>
      <div className = "mode-form sider-div">
        <h2> Select a game mode </h2>
        <form>
          <div className="radio">
          <label>
            <input type="radio" value="standard"
             checked={this.state.selectedOption === 'standard'}
              onChange= {this.handleOptionChange}/>
            Standard Chess
          </label>
        </div>
        <div className="radio">
          <label>
            <input type="radio" value="democrat"
            checked={this.state.selectedOption === 'democrat'}
            onChange= {this.handleOptionChange}/>
            Democrat
          </label>
        </div>
        <div className="radio">
          <label>
            <input type="radio" value="republican"
            checked={this.state.selectedOption === 'republican'}
            onChange= {this.handleOptionChange}/>
            Republican
          </label>
        </div>
        </form>
        <hr />
      </div>
      <div className = "message-div sider-div">
        <h2>{this.selectedPieceMessage()} </h2>
        <h2> {this.props.message} </h2>
      </div>
    </div>
  );
  }
};

export default ControlPanel;