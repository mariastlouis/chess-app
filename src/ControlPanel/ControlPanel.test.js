import React from 'react';
import ControlPanel from './ControlPanel.js';
import {shallow} from 'enzyme';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

const mockSetModeFunction = jest.fn();

let renderedControlPanel;

describe('ControlPanel', () => {
  beforeEach(() => {
    renderedControlPanel = shallow(
      <ControlPanel
        message = 'Select a piece to move'
        chessPiece = 'bishop'
        mode = {mockSetModeFunction} />);
  });

  it('should render correctly', () => {
    expect(renderedControlPanel).toBeDefined();
  });

  it('should match the snapshot', () =>{
    expect(renderedControlPanel).toMatchSnapshot();
  });


});