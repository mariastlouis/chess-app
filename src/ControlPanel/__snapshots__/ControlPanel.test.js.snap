// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlPanel should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ControlPanel
    chessPiece="bishop"
    message="Select a piece to move"
    mode={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="logo-div sider-div"
>
          <img
                    alt="logo"
                    className="logo"
                    src="logo.png"
          />
</div>,
        <div
          className="mode-form sider-div"
>
          <h2>
                     Select a game mode 
          </h2>
          <form>
                    <div
                              className="radio"
                    >
                              <label>
                                        <input
                                                  checked={true}
                                                  className="standard"
                                                  onChange={[Function]}
                                                  type="radio"
                                                  value="standard"
                                        />
                                        Standard Chess
                              </label>
                    </div>
                    <div
                              className="radio"
                    >
                              <label>
                                        <input
                                                  checked={false}
                                                  onChange={[Function]}
                                                  type="radio"
                                                  value="democrat"
                                        />
                                        Democrat
                              </label>
                    </div>
                    <div
                              className="radio"
                    >
                              <label>
                                        <input
                                                  checked={false}
                                                  onChange={[Function]}
                                                  type="radio"
                                                  value="republican"
                                        />
                                        Republican
                              </label>
                    </div>
          </form>
          <hr />
</div>,
        <div
          className="message-div sider-div"
>
          <h2>
                    Bishop selected
                     
          </h2>
          <h2>
                     
                    Select a piece to move
                     
          </h2>
</div>,
      ],
      "className": "controlPanel",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="logo"
            className="logo"
            src="logo.png"
/>,
          "className": "logo-div sider-div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "logo",
            "src": "logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
               Select a game mode 
</h2>,
            <form>
              <div
                            className="radio"
              >
                            <label>
                                          <input
                                                        checked={true}
                                                        className="standard"
                                                        onChange={[Function]}
                                                        type="radio"
                                                        value="standard"
                                          />
                                          Standard Chess
                            </label>
              </div>
              <div
                            className="radio"
              >
                            <label>
                                          <input
                                                        checked={false}
                                                        onChange={[Function]}
                                                        type="radio"
                                                        value="democrat"
                                          />
                                          Democrat
                            </label>
              </div>
              <div
                            className="radio"
              >
                            <label>
                                          <input
                                                        checked={false}
                                                        onChange={[Function]}
                                                        type="radio"
                                                        value="republican"
                                          />
                                          Republican
                            </label>
              </div>
</form>,
            <hr />,
          ],
          "className": "mode-form sider-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Select a game mode ",
            },
            "ref": null,
            "rendered": " Select a game mode ",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="radio"
>
                  <label>
                                    <input
                                                      checked={true}
                                                      className="standard"
                                                      onChange={[Function]}
                                                      type="radio"
                                                      value="standard"
                                    />
                                    Standard Chess
                  </label>
</div>,
                <div
                  className="radio"
>
                  <label>
                                    <input
                                                      checked={false}
                                                      onChange={[Function]}
                                                      type="radio"
                                                      value="democrat"
                                    />
                                    Democrat
                  </label>
</div>,
                <div
                  className="radio"
>
                  <label>
                                    <input
                                                      checked={false}
                                                      onChange={[Function]}
                                                      type="radio"
                                                      value="republican"
                                    />
                                    Republican
                  </label>
</div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <label>
                    <input
                                        checked={true}
                                        className="standard"
                                        onChange={[Function]}
                                        type="radio"
                                        value="standard"
                    />
                    Standard Chess
</label>,
                  "className": "radio",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        checked={true}
                        className="standard"
                        onChange={[Function]}
                        type="radio"
                        value="standard"
/>,
                      "Standard Chess",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "checked": true,
                        "className": "standard",
                        "onChange": [Function],
                        "type": "radio",
                        "value": "standard",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    "Standard Chess",
                  ],
                  "type": "label",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <label>
                    <input
                                        checked={false}
                                        onChange={[Function]}
                                        type="radio"
                                        value="democrat"
                    />
                    Democrat
</label>,
                  "className": "radio",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="radio"
                        value="democrat"
/>,
                      "Democrat",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "checked": false,
                        "onChange": [Function],
                        "type": "radio",
                        "value": "democrat",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    "Democrat",
                  ],
                  "type": "label",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <label>
                    <input
                                        checked={false}
                                        onChange={[Function]}
                                        type="radio"
                                        value="republican"
                    />
                    Republican
</label>,
                  "className": "radio",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        checked={false}
                        onChange={[Function]}
                        type="radio"
                        value="republican"
/>,
                      "Republican",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "checked": false,
                        "onChange": [Function],
                        "type": "radio",
                        "value": "republican",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    "Republican",
                  ],
                  "type": "label",
                },
                "type": "div",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Bishop selected
               
</h2>,
            <h2>
               
              Select a piece to move
               
</h2>,
          ],
          "className": "message-div sider-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Bishop selected",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Bishop selected",
              " ",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "Select a piece to move",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Select a piece to move",
              " ",
            ],
            "type": "h2",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo-div sider-div"
>
            <img
                        alt="logo"
                        className="logo"
                        src="logo.png"
            />
</div>,
          <div
            className="mode-form sider-div"
>
            <h2>
                         Select a game mode 
            </h2>
            <form>
                        <div
                                    className="radio"
                        >
                                    <label>
                                                <input
                                                            checked={true}
                                                            className="standard"
                                                            onChange={[Function]}
                                                            type="radio"
                                                            value="standard"
                                                />
                                                Standard Chess
                                    </label>
                        </div>
                        <div
                                    className="radio"
                        >
                                    <label>
                                                <input
                                                            checked={false}
                                                            onChange={[Function]}
                                                            type="radio"
                                                            value="democrat"
                                                />
                                                Democrat
                                    </label>
                        </div>
                        <div
                                    className="radio"
                        >
                                    <label>
                                                <input
                                                            checked={false}
                                                            onChange={[Function]}
                                                            type="radio"
                                                            value="republican"
                                                />
                                                Republican
                                    </label>
                        </div>
            </form>
            <hr />
</div>,
          <div
            className="message-div sider-div"
>
            <h2>
                        Bishop selected
                         
            </h2>
            <h2>
                         
                        Select a piece to move
                         
            </h2>
</div>,
        ],
        "className": "controlPanel",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="logo"
              className="logo"
              src="logo.png"
/>,
            "className": "logo-div sider-div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "logo",
              "src": "logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                 Select a game mode 
</h2>,
              <form>
                <div
                                className="radio"
                >
                                <label>
                                                <input
                                                                checked={true}
                                                                className="standard"
                                                                onChange={[Function]}
                                                                type="radio"
                                                                value="standard"
                                                />
                                                Standard Chess
                                </label>
                </div>
                <div
                                className="radio"
                >
                                <label>
                                                <input
                                                                checked={false}
                                                                onChange={[Function]}
                                                                type="radio"
                                                                value="democrat"
                                                />
                                                Democrat
                                </label>
                </div>
                <div
                                className="radio"
                >
                                <label>
                                                <input
                                                                checked={false}
                                                                onChange={[Function]}
                                                                type="radio"
                                                                value="republican"
                                                />
                                                Republican
                                </label>
                </div>
</form>,
              <hr />,
            ],
            "className": "mode-form sider-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Select a game mode ",
              },
              "ref": null,
              "rendered": " Select a game mode ",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="radio"
>
                    <label>
                                        <input
                                                            checked={true}
                                                            className="standard"
                                                            onChange={[Function]}
                                                            type="radio"
                                                            value="standard"
                                        />
                                        Standard Chess
                    </label>
</div>,
                  <div
                    className="radio"
>
                    <label>
                                        <input
                                                            checked={false}
                                                            onChange={[Function]}
                                                            type="radio"
                                                            value="democrat"
                                        />
                                        Democrat
                    </label>
</div>,
                  <div
                    className="radio"
>
                    <label>
                                        <input
                                                            checked={false}
                                                            onChange={[Function]}
                                                            type="radio"
                                                            value="republican"
                                        />
                                        Republican
                    </label>
</div>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <label>
                      <input
                                            checked={true}
                                            className="standard"
                                            onChange={[Function]}
                                            type="radio"
                                            value="standard"
                      />
                      Standard Chess
</label>,
                    "className": "radio",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          checked={true}
                          className="standard"
                          onChange={[Function]}
                          type="radio"
                          value="standard"
/>,
                        "Standard Chess",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "checked": true,
                          "className": "standard",
                          "onChange": [Function],
                          "type": "radio",
                          "value": "standard",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      "Standard Chess",
                    ],
                    "type": "label",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <label>
                      <input
                                            checked={false}
                                            onChange={[Function]}
                                            type="radio"
                                            value="democrat"
                      />
                      Democrat
</label>,
                    "className": "radio",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          checked={false}
                          onChange={[Function]}
                          type="radio"
                          value="democrat"
/>,
                        "Democrat",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "checked": false,
                          "onChange": [Function],
                          "type": "radio",
                          "value": "democrat",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      "Democrat",
                    ],
                    "type": "label",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <label>
                      <input
                                            checked={false}
                                            onChange={[Function]}
                                            type="radio"
                                            value="republican"
                      />
                      Republican
</label>,
                    "className": "radio",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          checked={false}
                          onChange={[Function]}
                          type="radio"
                          value="republican"
/>,
                        "Republican",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "checked": false,
                          "onChange": [Function],
                          "type": "radio",
                          "value": "republican",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      "Republican",
                    ],
                    "type": "label",
                  },
                  "type": "div",
                },
              ],
              "type": "form",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Bishop selected
                 
</h2>,
              <h2>
                 
                Select a piece to move
                 
</h2>,
            ],
            "className": "message-div sider-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Bishop selected",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Bishop selected",
                " ",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "Select a piece to move",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Select a piece to move",
                " ",
              ],
              "type": "h2",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
